app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -82,17 +84,18 @@ private void onClickByCity(View view) {
             public void run() {
                 Log.d(TAG, "run started");
 
+                Map<String, String> weather;
```

Comment:
The idea to collect all weather info in single object is correct, but do not use map for it.
Create simple class Weather {
private int temperature;
private int humidity;
...
getters & setters
}